src/inference/src/dev/core_impl.cpp

```diff
@@ -64,6 +64,13 @@ void stripDeviceName(std::string& device, const std::string& substr) {
     }
 }
 
+void validDeviceName(const std::string& device) {
+    auto pos = device.rfind(")");
+    if (pos != std::string::npos && pos != device.length() - 1) {
```

Comment:
We should check any possible characters in device name are after all parenthesis, not only the last parenthesis.